Support arm64 coroutines.

Set correct shared library name.

Override the arch setting to remove OpenBSD version from it,
so ports don't have to be bumped when OpenBSD version changes.

Index: configure
--- configure.orig
+++ configure
@@ -29533,6 +29533,10 @@ esac
 
             coroutine_type=x86
          ;; #(
+  aarch64-openbsd*) :
+
+            coroutine_type=arm64
+         ;; #(
   *-openbsd*) :
 
             coroutine_type=pthread
@@ -31793,7 +31797,7 @@ fi
 	 ;; #(
   openbsd*|mirbsd*) :
 
-	LIBRUBY_SO='lib$(RUBY_SO_NAME).$(SOEXT).$(MAJOR).'`expr ${MINOR} \* 10 + ${TEENY}`
+	LIBRUBY_SO='lib$(RUBY_SO_NAME).so.'${LIBruby31_VERSION}
 	 ;; #(
   solaris*) :
 
@@ -33495,7 +33499,7 @@ then :
 
 else $as_nop
 
-        arch="${target_cpu}-${target_os}"
+        arch="${target_cpu}-openbsd"
 
 fi
     printf "%s\n" "#define RUBY_PLATFORM \"$arch\"" >>confdefs.h
